Makefile

```diff
@@ -94,6 +94,7 @@ clean:
 	rm -f $(MICROSERVICES)
 
 test:
+	docker run --rm -v `pwd`:/host:ro,z --entrypoint /bin/hadolint hadolint/hadolint:latest --config /host/.hadolint.yml `find * -name 'Dockerfile*' | xargs -i echo '/host/{}'` 
```

Comment:
I assume this will run fine in the CI pipeline?

Reply:
Nope, it doesn't... ;-) See failure below.

Reply:
Was worth a shot.

Reply:
Next try of this will see if docker exists before running the command.

@ernestojeda It looks like the `make test` takes place inside a container (because the pipeline uses docker for plenty of other things).  What do you think is the best way to structure this such that it runs in the pipeline, but can also run outside of the pipeline as well?

Reply:
I could also try to download hadolint directly from GitHub (https://github.com/hadolint/hadolint/releases/tag/v2.6.0) but then I need to detect if I am in Windows or Linux.

Reply:
We have a ci docker image: https://github.com/edgexfoundry/ci-build-images/blob/golang-1.16/Dockerfile where we could add that hadolint binary so it would be available in the image. That's probably the easiest thing to do.

Reply:
@ernestojeda are you proposal we do this?

if which hadolint; then
   run hadolint directly
 else
   run hadolint via docker
 fi

Reply:
Yeah, that would work. I will get hadolint in the CI image now and we can test this.

Reply:
@ernestojeda ok. pushed an updated conditional.  Please give it a shot.

Reply:
PS. To test it remove some of the ignores in the config file.